src/materialized/tests/sql.rs

```diff
@@ -943,3 +943,89 @@ write frontier:[            0]\n",
 
     Ok(())
 }
+
+#[test]
+fn test_linearizability() -> Result<(), Box<dyn Error>> {
```

Comment:
@mjibson btw, this is deterministically failing for me locally. I just can't seem to get it to run in CI.

Reply:
I'm not sure what the error message means. Try rebasing this on top of #12544? I'm not sure what the thing is that's failing to connect with "Cannot assign requested address". The trace says it's from `mz_dataflow_types::client::tcp` which is maybe a compute/storage thing? Is it possible the POSTGRES_URL is in some unexpected format or something?